### PR TITLE
4018 Error 403 in development environment when uploading files to S3

### PR DESCRIPTION
**Pull request by blancoramiro**
This PR is related to issue #4018 

- Removes unused apparently unused AWS variables from settings
- Explicitly sets AWS environment variables in courtlistener docker compose file.

### CODE REVIEW

**Comment by quevon24** [comment]
@mlissner the change looks good, it is easy to use, we just need to paste the environment variables provided by aws (there is a copy button in the portal) into the host terminal:

```
export AWS_ACCESS_KEY_ID="xxxx"
export AWS_SECRET_ACCESS_KEY="xxxx"
export AWS_SESSION_TOKEN="xxxx"
```

and then update the env variables in the container:

`docker-compose -f docker/courtlistener/docker-compose.yml up -d
`

As the keys expire every 8 hours, this would be necessary to do it daily only if you are going to use s3 file upload

File: cl/settings/third_party/aws.py [hunk 1]
```diff
@@ -3,15 +3,7 @@
...
 DEVELOPMENT = env.bool("DEVELOPMENT", default=True)
 
-
 # S3
-if DEVELOPMENT:
-    AWS_ACCESS_KEY_ID = env("AWS_DEV_ACCESS_KEY_ID", default="")
-    AWS_SECRET_ACCESS_KEY = env("AWS_DEV_SECRET_ACCESS_KEY", default="")
-else:
-    AWS_ACCESS_KEY_ID = env("AWS_ACCESS_KEY_ID", default="")
-    AWS_SECRET_ACCESS_KEY = env("AWS_SECRET_ACCESS_KEY", default="")
```
Comment (mlissner) [inline cl/settings/third_party/aws.py:13]: Hm, these are used in prod. Are you sure we can remove them safely?

Comment (blancoramiro) [reply cl/settings/third_party/aws.py:13]: I am pretty sure boto is picking them up from the environment vars.
But just to be sure I just reverted this change. Too risky.

**Comment by mlissner** [comment]
A few notes to get this over the line, since it's one of your first CourtListener PRs, @blancoramiro:

1. We need you to sign the contributor thing (see comment above).
2. Tests aren't passing yet, when they're fixed, please mark it as ready for review (it's still a draft), and you can set me as the reviewer.

File: cl/settings/third_party/aws.py (hunk 1 above)
Comment (quevon24) [reply cl/settings/third_party/aws.py:13]: > Hm, these are used in prod. Are you sure we can remove them safely?

mmm that is a good question, we can probably leave it as I suggested at https://github.com/freelawproject/courtlistener/issues/4018#issue-2277950289

**Comment by mlissner** [comment]
Should we put this in the dev documentation?

**Comment by blancoramiro** [comment]
Yes. Probably [here](https://github.com/freelawproject/courtlistener/wiki/Getting-Started-Developing-CourtListener#developer-installation)

Also not sure if it's better to receive a warning when you run the builds or if we want to setup some default bogus values for these variables.


```
WARNING: The AWS_ACCESS_KEY_ID variable is not set. Defaulting to a blank string.
WARNING: The AWS_SECRET_ACCESS_KEY variable is not set. Defaulting to a blank string.
WARNING: The AWS_SESSION_TOKEN variable is not set. Defaulting to a blank string.
```

Maybe something like this:

```
    environment:
      - AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID:-readdocumentation}
      - AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY:-readdocumentation}
      - AWS_SESSION_TOKEN=${AWS_SESSION_TOKEN:-readdocumentation}
```

**Comment by mlissner** [comment]
¿Porqúe no los dos? We can do warnings and also make the values a pointer to the documentation. Hell, you could make them bit.ly links or something, no? 

> Yes. Probably [here](https://github.com/freelawproject/courtlistener/wiki/Getting-Started-Developing-CourtListener#developer-installation)

Looks right to me.

**Comment by blancoramiro** [comment]
Sadly it looks like it's one or the other, at least with out adding something else like a script that checks the vars before docker-compose. There might be a way that I am missing though.
[Interpolation](https://docs.docker.com/compose/environment-variables/env-file/#interpolation)

I am thinking that it'll be best to remove the warning since it might lead contributors to think that they *need* those in place to work when probably not.

Added [this](https://github.com/freelawproject/courtlistener/wiki/Getting-Started-Developing-CourtListener#aws) to the documentation. I would appreciate a review, not a 100% happy with it.

Do we habe a bit.ly account? I think it's safe to add the full url, but maybe not as pretty.

**Comment by mlissner** [comment]
Eh, let's do the full URL and forget bit.ly, that's probably better anyway.

If we have to choose one, removing the warning seems fine to me too. 

I'm not a fan of having to set and export env's. Don't we usually set things in our .env file?

**Comment by blancoramiro** [comment]
> Eh, let's do the full URL and forget bit.ly, that's probably better anyway.
> If we have to choose one, removing the warning seems fine to me too.

I'll create the PR.

> I'm not a fan of having to set and export env's. Don't we usually set things in our .env file?

The issue is that now since we are using the SSO and using temporary credentials adding them to the .env file could be more troublesome than just login, copying the creds with the `copy` button, pasting them into the terminal and then running docker compose.

The SSO provides an easy way copy the credentials ready to be pasted and export the variables:

creds


Plus this way no keys are stored on disk.

**Comment by mlissner** [comment]
Got it!

**Pull request merged**